client/src/utils/components/list/List.tsx

```diff
@@ -0,0 +1,125 @@
+/*!
+ * Copyright 2022 - Swiss Data Science Center (SDSC)
+ * A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+ * Eidgenössische Technische Hochschule Zürich (ETHZ).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as React from "react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faGlobe, faLock, faPlay, faPlus, faUserFriends } from "@fortawesome/free-solid-svg-icons";
+import "./ListCard.css";
+import { toCapitalized } from "../../helpers/HelperFunctions";
+import { Visibilities } from "../visibility/Visibility";
+import { Button } from "../../ts-wrappers";
+import { EntityType, ListProps, ListType } from "./List.d";
+import ListCard from "./ListCard";
+import ListBar from "./ListBar";
+
+export interface VisibilityIconProps {
+  visibility: Visibilities;
+}
+
+export const VisibilityIcon = ({ visibility }: VisibilityIconProps) => {
+  if (!visibility) return null;
+  const icon = {
+    public: <FontAwesomeIcon icon={faGlobe} />,
+    private: <FontAwesomeIcon icon={faLock} />,
+    internal: <FontAwesomeIcon icon={faUserFriends} />
+  };
+  return <span className="visibility-icon text-rk-text" title={toCapitalized(visibility)}>
+    { icon[visibility] || "" }
+  </span>;
+};
+
+/**
+ * ListCard/ListBar returns a card or a bar displaying an item in a List.
+ *
+ * @param type Type of visualization (Card, Bar).
+ * @param url containing a link to the entity details.
+ * @param title title of the entity.
+ * @param slug slug of the entity.
+ * @param description description of the entity.
+ * @param visibility visibility of the entity.
+ * @param tagList tag list of the entity.
+ * @param timeCaption date to put inside the time caption of the entity.
+ * @param labelCaption label to put inside the time caption of the entity, if empty defaults to Updated.
+ * @param mediaContent image of the entity.
+ * @param creators creators of the entity, if more than 3 they will be cropped at 3.
+ * @param itemType type of the entity being rendered, the color of the circle depends on this.
+ * @param handler handler function for main button.
+ */
+function List(
+  { type,
+    url,
+    title,
+    description,
+    tagList,
+    timeCaption,
+    labelCaption,
+    mediaContent,
+    creators,
+    itemType,
+    slug,
+    visibility,
+    handler
+  }: ListProps) {
```

Comment:
I suggest changing this, as it will make things below a little less complicated.

```suggestion
function List(props: ListProps) {
const { type,
    url,
    title,
    description,
    tagList,
    timeCaption,
    labelCaption,
    mediaContent,
    creators,
    itemType,
    slug,
    visibility,
    handler
  } = props;
```